src/java.desktop/share/native/libharfbuzz/hb-meta.hh

```diff
@@ -188,7 +188,7 @@ template <> struct hb_int_max<signed long long>         : hb_integral_constant<s
 template <> struct hb_int_max<unsigned long long>       : hb_integral_constant<unsigned long long,      ULLONG_MAX>     {};
 #define hb_int_max(T) hb_int_max<T>::value
 
-#if defined(__GNUC__) && __GNUC__ < 5
+#if defined(__GNUC__) && __GNUC__ < 5 && !defined(__clang__)
```

Comment:
Normally, such changes in third-party libraries need to be done upstream, and not locally. @prrace can confirm.

Reply:
> Normally, such changes in third-party libraries need to be done upstream, and not locally. @prrace can confirm.

Thanks @kevinrushforth for your review.

Yes, it had been fixed in the upstream and I just follow it.
Please see https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-meta.hh#L202 .

Or did you mean we had to sync the whole harfbuzz with the upstream?
If so, I would suggest doing it in a separate PR.
What do you think?
Thanks.

Reply:
If the *exact* same change has been committed upstream then we are OK to take this because then when we upgrade harfbuzz (under a separate bug) whoever doing the upgrade can be blissfully ignorant of this. In fact it is because there is no easy way to be track changes on the JDK side and reapply them that we have this rule.